include/alpaka/mem/buf/oacc/Copy.hpp

```diff
@@ -0,0 +1,395 @@
+                Vec<TDim, SrcSize> const m_srcExtent;
+#endif
+                std::uint8_t * const m_dstMemNative;
+                std::uint8_t const * const m_srcMemNative;
+                TCopyPred m_copyPred;
+
+            };
+
+            //#############################################################################
+            //! The OpenAcc Nd device memory copy task.
+            //!
+            template<
+                typename TDim,
+                typename TViewDst,
+                typename TViewSrc,
+                typename TExtent,
+                typename TCopyPred>
+            struct TaskCopyOacc : public TaskCopyOaccBase<TDim, TViewDst, TViewSrc, TExtent, TCopyPred>
+            {
+                using DimMin1 = DimInt<TDim::value - 1u>;
+                using typename TaskCopyOaccBase<TDim, TViewDst, TViewSrc, TExtent, TCopyPred>::ExtentSize;
+                using typename TaskCopyOaccBase<TDim, TViewDst, TViewSrc, TExtent, TCopyPred>::DstSize;
+                using typename TaskCopyOaccBase<TDim, TViewDst, TViewSrc, TExtent, TCopyPred>::SrcSize;
+
+                //-----------------------------------------------------------------------------
+                using TaskCopyOaccBase<TDim, TViewDst, TViewSrc, TExtent, TCopyPred>::TaskCopyOaccBase;
+
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST auto operator()() const
+                -> void
+                {
+                    ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                    this->printDebug();
+#endif
+                    Vec<DimMin1, ExtentSize> const extentWithoutInnermost(subVecBegin<DimMin1>(this->m_extent));
+                    // [z, y, x] -> [y, x] because the z pitch (the full size of the buffer) is not required.
+                    Vec<DimMin1, DstSize> const dstPitchBytesWithoutOutmost(subVecEnd<DimMin1>(this->m_dstPitchBytes));
+                    Vec<DimMin1, SrcSize> const srcPitchBytesWithoutOutmost(subVecEnd<DimMin1>(this->m_srcPitchBytes));
+
+                    if(static_cast<std::size_t>(this->m_extent.prod()) != 0u)
+                    {
+                        this->m_dev.makeCurrent();
+                        meta::ndLoopIncIdx(
+                            extentWithoutInnermost,
+                            [&](Vec<DimMin1, ExtentSize> const & idx)
+                            {
+                                this->m_copyPred(
+                                    reinterpret_cast<void *>(
+                                        this->m_dstMemNative + (castVec<DstSize>(idx) * dstPitchBytesWithoutOutmost).foldrAll(std::plus<DstSize>())),
+                                    const_cast<void*>(reinterpret_cast<const void *>(
+                                        this->m_srcMemNative + (castVec<SrcSize>(idx) * srcPitchBytesWithoutOutmost).foldrAll(std::plus<SrcSize>()))),
```

Comment:
There is a shortcut for `foldrAll(plus)` called `sum()`.

Reply:
Did not notice this member, thanks.